app/actions/AuthenticatedAction.scala

```diff
@@ -16,62 +16,56 @@
 
 package actions
 
+import javax.inject.Inject
+
 import auth.GovernmentGatewayProvider
 import config.{ApplicationConfig, Global}
 import connectors._
-import javax.inject.Inject
-import models.{DetailedIndividualAccount, GroupAccount}
+import models.{Accounts, DetailedIndividualAccount, GroupAccount}
 import play.api.Logger
 import play.api.i18n.{I18nSupport, Messages, MessagesApi}
-import play.api.libs.concurrent.Execution.Implicits.defaultContext
 import play.api.mvc.Results._
 import play.api.mvc._
+import services.{EnrolmentResult, EnrolmentService, Failure, Success}
 import uk.gov.hmrc.auth.core._
-import uk.gov.hmrc.http.{BadRequestException, HeaderCarrier, NotFoundException}
+import uk.gov.hmrc.auth.core.retrieve.Retrievals._
+import uk.gov.hmrc.auth.core.retrieve.~
+import uk.gov.hmrc.http.HeaderCarrier
 import uk.gov.hmrc.play.HeaderCarrierConverter
 
-import scala.concurrent.Future
+import scala.concurrent.{ExecutionContext, Future}
 
-class AuthenticatedAction @Inject()(
-                                     provider: GovernmentGatewayProvider,
-                                     businessRatesAuthorisation: BusinessRatesAuthorisation,
-                                     auth: Auth,
-                                     addressesConnector: Addresses,
+class AuthenticatedAction @Inject()(override val messagesApi: MessagesApi,
+                                    provider: GovernmentGatewayProvider,
+                                    businessRatesAuthorisation: BusinessRatesAuthorisation,
+                                    enrolmentService: EnrolmentService,
                                     val authConnector: AuthConnector
-                                   )(implicit val messageApi: MessagesApi, config: ApplicationConfig) extends I18nSupport{
+                                   )(implicit val messageApi: MessagesApi, config: ApplicationConfig, executionContext: ExecutionContext)
+  extends ActionBuilder[BasicAuthenticatedRequest] with AuthorisedFunctions with I18nSupport{
 
   val logger = Logger(this.getClass.getName)
 
-  override def messagesApi: MessagesApi = messageApi
-
   protected implicit def hc(implicit request: Request[_]): HeaderCarrier = HeaderCarrierConverter.fromHeadersAndSession(request.headers, Some(request.session))
 
-  def apply(body: BasicAuthenticatedRequest[AnyContent] => Future[Result]) = Action.async { implicit request =>
+  override def invokeBlock[A](request: Request[A], block: BasicAuthenticatedRequest[A] => Future[Result]): Future[Result] = {
+    implicit val hc: HeaderCarrier = HeaderCarrierConverter.fromHeadersAndSession(request.headers, Some(request.session))
+    logger.debug("the request called invoke block")
     businessRatesAuthorisation.authenticate flatMap {
```

Comment:
@DanielConnelly 
i thought we were trying to remove BRA ?

Reply:
Might do separate Jira ticket , already we have done lot of changes to part of this ticket